crates/aptos/src/config/mod.rs

```diff
@@ -291,3 +308,39 @@ impl FromStr for ConfigType {
         }
     }
 }
+
+const ASSUME_YES: &str = "yes";
+const ASSUME_NO: &str = "no";
+
+/// A configuration for how to expect the prompt response
+///
+/// `true` will tail `--assume-yes` option
+/// `false` will tail `--assume-no` option
+#[derive(Debug, Copy, Clone, Serialize, Deserialize)]
+pub struct PromptResponseType(bool);
+
+impl Default for PromptResponseType {
+    fn default() -> Self {
+        Self(true)
+    }
+}
+
+impl std::fmt::Display for PromptResponseType {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        f.write_str(if self.0 { ASSUME_YES } else { ASSUME_NO })
+    }
+}
+
+impl FromStr for PromptResponseType {
+    type Err = CliError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.to_lowercase().trim() {
+            ASSUME_YES => Ok(Self(true)),
+            ASSUME_NO => Ok(Self(false)),
+            _ => Err(CliError::CommandArgumentError(
+                "Invalid prompt response type, must be one of [yes, no]".to_string(),
+            )),
+        }
+    }
```

Comment:
You may want a third option `PROMPT` or `CLEAR` which gets rid of this defaulting, and goes back to the original prompt method afterwards (and basically just removes this field